src/js/components/Button/Button.js

```diff
@@ -281,8 +322,17 @@ const Button = forwardRef(
             theme.global.colors.text[isDarkBackground() ? 'dark' : 'light'],
         });
       }
+    } else if (kind && kindIcon && !plain) {
+      const iconColor =
+        (hover && getIconColor(themePaths.hover, theme)) ||
+        getIconColor(themePaths.base, theme, color, kind);
+      if (iconColor)
+        buttonIcon = cloneElement(kindIcon, {
+          color: iconColor,
+        });
     }
 
+    const reverse = reverseProp || theme.button[kind]?.reverse;
```

Comment:
If `reverseProp` is `false`, this will use the theme prop instead of allowing the property to override the theme. I think maybe:
```suggestion
    const reverse = reverseProp ?? theme.button[kind]?.reverse;
```